sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -76,13 +80,95 @@ def assert_argkmin_results_equality(ref_dist, dist, ref_indices, indices):
         ref_dist,
         dist,
         err_msg="Query vectors have different neighbors' distances",
-        rtol=1e-7,
+        rtol=rtol,
     )
 
 
-def assert_radius_neighborhood_results_equality(ref_dist, dist, ref_indices, indices):
+def adaptive_rounding(scalar, n_significant_digits):
+    """Round a scalar to a number of significant digits adaptively to its value."""
+    magnitude = int(floor(log10(abs(scalar)))) + 1
+    return round(scalar, n_significant_digits - magnitude)
+
+
+def test_adaptive_rounding():
+
+    assert adaptive_rounding(123456789, 2) == 120000000
+    assert adaptive_rounding(123456789, 3) == 123000000
+    assert adaptive_rounding(123456789, 10) == 123456789
+
+    assert adaptive_rounding(1.23456789, 2) == 1.2
+    assert adaptive_rounding(1.23456789, 3) == 1.23
+    assert adaptive_rounding(1.23456789, 10) == 1.23456789
+
+
+def assert_argkmin_results_quasi_equality(
+    ref_dist,
+    dist,
+    ref_indices,
+    indices,
+    rtol=1e-4,
+    decimals=5,
```

Comment:
I think `decimals` should be renamed to either `n_significant_digits` or `sig_digits` to convey that this is a relative way of specifying a precision level.

Then would it make sense to automatically define `rtol` to `10 ** -sig_digits` or `10 ** (1 - sig_digits)`? I am not sure. Maybe it's more flexible to heep the 2 parameter independent.

Reply:
The second part is a duplicate of https://github.com/scikit-learn/scikit-learn/pull/23490#discussion_r892266420.

Reply:
Slight preference toward keeping rtol, but as you prefer